core/model/modx/registry/modfileregister.class.php

```diff
@@ -263,4 +272,14 @@ public function send($topic, $message, array $options = array()) {
     public function close() {
         return true;
     }
+
+    /**
+     * Sanitize the specified path
+     *
+     * @param string $path The path to clean
+     * @return string The sanitized path
+     */
+    protected function sanitizePath($path) {
+        return preg_replace(array("/\.*[\/|\\\]/i", "/[\/|\\\]+/i"), array('/', '/'), $path);
```

Comment:
Any reason the two expressions can not be merged to `/\.*[\/|\\\]+/i`? It should match both `../`, `..////` and `////`. You might also want to consider an expression such as `(?:\.*[\/|\\]+)+`. This matches `../../....//////` too.